packages/datadog-plugin-pg/test/index.spec.js

```diff
@@ -396,24 +397,37 @@ describe('Plugin', () => {
         })
 
         it('query text should contain traceparent', done => {
-          let queryText = ''
+          let seenTraceParent
+          let seenTraceId
+          let seenSpanId
+          const originalWrite = net.Socket.prototype.write
+          net.Socket.prototype.write = function (buffer) {
+            let strBuf = buffer.toString()
+            if (strBuf.includes('traceparent=\'')) {
+              strBuf = strBuf.split('-')
+              seenTraceParent = true
+              seenTraceId = strBuf[1]
+              seenSpanId = strBuf[2]
+            }
+            return originalWrite.apply(this, arguments)
+          }
+
           agent.use(traces => {
             const traceId = traces[0][0].trace_id.toString(16).padStart(32, '0')
             const spanId = traces[0][0].span_id.toString(16).padStart(16, '0')
-
-            expect(queryText).to.equal(
-              `/*dddbs='post',dde='tester',ddps='test',ddpv='8.4.0',` +
-              `traceparent='00-${traceId}-${spanId}-00'*/ SELECT $1::text as message`)
+            expect(seenTraceId).to.equal(traceId)
+            expect(seenSpanId).to.equal(spanId)
           }).then(done, done)
 
           client.query('SELECT $1::text as message', ['Hello World!'], (err, result) => {
             if (err) return done(err)
+            expect(seenTraceParent).to.be.true
+            net.Socket.prototype.write = originalWrite
```

Comment:
This will not get restored if the assertion before it fails. This could cause other tests to fail incorrectly. It's generally best to avoid side effects potentially leaking out of tests. This could be resolved by putting the test in a `describe` with `before` and `after` to do the environment modifications, ensuring they always happen regardless of what happens in the test.

```js
describe('something', () => {
  before(() => {
    // Do your environment-changing things...
  })

  after(() => {
    // Ensure your environment changes are restored, even if the tests failed.
  })

  it('should work', () => {
    // Your modified environment state will be present here.
  })
})
```

Reply:
Got it, good point, will make the changes 👍

Reply:
Hi @Qard I have added the tests to a beforeEach and and afterEach block.